File: BZSwift/1222test/1222test/ViewController.m
//
//  ViewController.m
//  1222test
//
//  Created by ZBin on 15/12/23.
//  Copyright © 2015年 ZBin. All rights reserved.
//

#import "ViewController.h"
#import <UIKit/UIGestureRecognizer.h>
@interface ViewController ()

@end

@implementation ViewController


- (void)viewDidLoad {
    UISwipeGestureRecognizer* recognizer;
    // handleSwipeFrom 是偵測到手势，所要呼叫的方法
    recognizer = [[UISwipeGestureRecognizer alloc] initWithTarget:selfaction:@selector(handleSwipeFrom)];
    // 不同的 Recognizer 有不同的实体变数
    // 例如 SwipeGesture 可以指定方向
    // 而 TapGesture 則可以指定次數
    recognizer.direction = UISwipeGestureRecognizerDirectionUp
    [self.view addGestureRecognizer:recognizer];
    [recognizer release];
}

- (void)handleSwipeFrom:(UISwipeGestureRecognizer*)recognizer {
    // 触发手勢事件后，在这里作些事情
    // 底下是刪除手势的方法
    [self.view removeGestureRecognizer:recognizer];
}


@end

File: BZSwift/1225test/1225test/ViewController.swift
//
//  ViewController.swift
//  1225test
//
//  Created by ZBin on 15/12/26.
//  Copyright © 2015年 ZBin. All rights reserved.
//

import UIKit

class ViewController: UIViewController {
    
    var button: UIButton!
    var progressView: UIProgressView!
    
    var timer: NSTimer!
    var remainTime = 100
    
    override func viewDidLoad() {
        
        // 初始化按钮，开始倒计时
        button = UIButton(type: .System) as UIButton
        button.frame = CGRectMake(self.view.frame.width/2 - 50, 50, 100, 50)
        button.setTitle("开始", forState: UIControlState.Normal)
        
        button.addTarget(self, action: "buttonAction", forControlEvents: UIControlEvents.TouchUpInside)
        
        // 初始化 progressView
        progressView = UIProgressView(progressViewStyle: UIProgressViewStyle.Bar)
        progressView.frame = CGRectMake(self.view.frame.width/2 - 50, 200, 100, 100)
        // 设置初始值
        progressView.progress = 1.0
        // 设置进度条颜色
        progressView.progressTintColor = UIColor.blueColor()
        // 设置进度轨迹颜色
        progressView.trackTintColor = UIColor.greenColor()
        // 扩展：可以通过 progressImage、trackImage 属性自定义出个性进度条
        
        self.view.addSubview(button)
        self.view.addSubview(progressView)
    }
    
    /// 响应按钮点击事件，开始倒计时
    func buttonAction() {
        button.enabled = false
        timer = NSTimer.scheduledTimerWithTimeInterval(1, target: self, selector: "timerAction", userInfo: nil, repeats:true)
        timer.fire()
    }
    
    // 每秒定时触发
    func timerAction() {
        if(remainTime < 0){
            //倒计时结束
            timer.invalidate()
        } else {
            print("\(remainTime)")
            remainTime = remainTime - 1
            let progressValue = Float(remainTime)/100
            progressView.setProgress(progressValue, animated:true)
        }
    }
    
}

